src/Oscoin/API/HTTP.hs

```diff
@@ -20,23 +23,66 @@ import qualified Network.Wai.Middleware.Static as Wai
 
 -- TODO: Don't import this here? Create a HTTP.Routing module?
 import qualified Network.Wai as Wai
+import           Web.HttpApiData (FromHttpApiData(..))
 import           Web.Spock
 
 import           Codec.Serialise (Serialise)
-import           Data.Aeson (ToJSON)
+import           Data.Aeson (FromJSON, ToJSON)
+import           Formatting.Buildable (Buildable)
 
-withTelemetryMiddleware :: Node.Handle tx e s i
+withTelemetryMiddleware :: Node.Handle c tx e s i
                         -> (Wai.Middleware -> IO a)
                         -> IO a
 withTelemetryMiddleware hdl action = do
     let handle = Node.cfgTelemetry . Node.Trans.hConfig $ hdl
     action (telemetryMiddleware handle)
 
-run :: (ToJSON s, Serialise s) => Int -> Node.Handle RadTx Rad.Env s i -> IO ()
+run
+    :: ( ToJSON (Sealed c s)
+       , ToJSON (Crypto.PK c)
+       , Typeable c
+       , FromHttpApiData (BlockHash c)
+       , Serialise s
+       , Serialise (BlockHash c)
+       , Serialise (Crypto.PK c)
+       , Serialise (Crypto.Signature c)
+       , Crypto.HasDigitalSignature c
+       , Crypto.HasHashing c
+       , FromJSON (BlockHash c)
+       , FromJSON (Crypto.PK c)
+       , FromJSON (Crypto.Signature c)
+       , Ord (Crypto.Hash c)
+       , Buildable (Crypto.Hash c)
+       , ToJSON (Crypto.Hash c)
+       , ToJSON (Crypto.Signature c)
+       )
+    => Int
+    -> Node.Handle c (RadTx c) (Rad.Env c) s i
+    -> IO ()
```

Comment:
Do we ever need to run this node with non-`RealWorldCrypto`?

Reply:
Do we (or do we plan to) exercise this part of the codebase via the testsuite? If this is a top-level runner we use in the executable only we can monomorphise this over `RealWorldCrypto`.

Reply:
Currently we have a separate node for testing (`TestNode`) - so I would monomorphise for now

Reply:
cf21232